DrinkManagerWeb/Views/Drinks/Partials/DrinkList.cshtml

```diff
@@ -1,15 +1,36 @@
-﻿@model DrinkManagerWeb.Models.ViewModels.DrinksViewModel
+﻿@using BLL.Data.Repositories
+@using System.Security.Claims
+@model DrinkManagerWeb.Models.ViewModels.DrinksViewModel
+@inject IFavouriteRepository _favourite
 
 <div class="row">
     @foreach (var drink in Model.Drinks.OrderBy(x => x.Name))
     {
         <div class="col-lg-3 col-md-6 col-sm-12 mb-3">
             <div class="card">
-                <a class="page-link" asp-action="DrinkDetails" asp-route-id="@drink.DrinkId"><img class="w-100 rounded shadow-lg" src="@drink.ImageUrl" /></a>
+                @if (_favourite.IsFavourite(User.FindFirstValue(ClaimTypes.NameIdentifier), drink.DrinkId))
+                {
+                    <i class="fa fa-heart heart"></i>
+                }
+                @if (drink.DrinkReviews != null && drink.DrinkReviews.Any())
+                {
+
+                    var averageReview = drink.DrinkReviews.Select(x => x.ReviewScore).Average();
+                    var i = Convert.ToInt32(averageReview);
```

Comment:
Such logic should not be done inside a view, maybe we could create an AverageReview Property for the drink?